layer/clients/model_catalog.py

```diff
@@ -170,53 +177,83 @@ def create_model_train_from_version_id(
         )
         return response.id
 
-    def load_by_model_definition(
+    def load_model_artifact(
         self,
-        model_definition: ModelDefinition,
+        model: Model,
+        state: ResourceTransferState,
         no_cache: bool = False,
-        state: Optional[ResourceTransferState] = None,
-    ) -> Any:
+    ) -> ModelArtifact:
         """
-        Loads a model from the model catalog
+        Loads a model artifact from the model catalog
 
-        :param model_definition: model definition
-        :return: a model object
+        :param model: the model
```

Comment:
might as well update the docs for other params too = )

Reply:
saving this for future cleanup of docstrings